level.go

```diff
@@ -93,10 +93,19 @@ func NewLRU(maxEntries int) *LRU {
 
 // Add adds the entry to the cache.
 func (l *LRU) Add(_ context.Context, k Key, e *Entry, ttl time.Duration) error {
+	oldBytes, marshalErr:= e.MarshalBinary()
+	if marshalErr!= nil{
+		return marshalErr
+	}
+	newEntry := &Entry{}
+	unmarshalErr := newEntry.UnmarshalBinary(oldBytes)
+	if unmarshalErr != nil {
+		return unmarshalErr
+	}
```

Comment:
```suggestion
	buf, err := e.MarshalBinary()
	if err != nil {
		return err
	}
	newEntry := &Entry{}
	if  err := newEntry.UnmarshalBinary(buf); err != nil {
		return err
	}
```

Looks good, just minor style suggestions.